### PR TITLE
[BOT] Add ci-no-td to reland PRs

### PR DESCRIPTION
**Pull request by yangw-dev**
Issue: https://github.com/pytorch/pytorch/issues/139492

# Overview
Add label `ci-no-td` to reland PullRequest.

# Notice
I split the list for pr and issue regex, this makes us to hand labels in different situations.

### CODE REVIEW

File: torchci/lib/bot/autoLabelBot.ts [hunk 1]
```diff
@@ -283,13 +296,28 @@ function isNotUserFacing(filesChanged: string[]): boolean {
...
 
-  for (const [regex, label] of titleRegexToLabel) {
+  for (const [regex, label] of IssueTitleRegexToLabel) {
+    if (title.match(regex) && label.match(labelFilter)) {
+      labelsToAdd.push(label);
+    }
+  }
+
+  return labelsToAdd;
+}
```
Comment (clee2000) [inline torchci/lib/bot/autoLabelBot.ts:312]: Maybe a helper function or something that takes in a list to reduce code duplication?

File: torchci/lib/bot/autoLabelBot.ts [hunk 2]
```diff
@@ -12,7 +12,20 @@ import {
...
+// List of regex patterns for assigning labels to Pull Requests
+const PrTitleRegexToLabel: [RegExp, string][] = [
+  [/reland/gi, "ci-no-td"],
+  [/rocm/gi, "module: rocm"],
+  [/rocm/gi, "ciflow/rocm"],
+  [/vulkan/gi, "module: vulkan"],
+  [/vulkan/gi, "ciflow/periodic"], // Vulkan tests are run periodically
+  [/DISABLED\s+test.*\(.*\)/g, "skipped"],
+  [/UNSTABLE\s+.*\s+\/\s+.*/g, "unstable"],
+  [/UNSTABLE\s+.*\s+\/\s+.*/g, "module: ci"],
```
[the remark this answers is not in the document]
Comment (clee2000) [reply torchci/lib/bot/autoLabelBot.ts:24]: Also, ciflow/periodic probably shouldn't be in IssueTitleRegexToLabel, and DISABLED, UNSTABLE shouldn't be in PrTitleRegexToLabel

[the remark this answers is not in the document]
Comment (huydhn) [reply torchci/lib/bot/autoLabelBot.ts:24]: I killed vulkan job in periodic recently as edge team doesn't need it anymore, we can safely ignore that one

[the remark this answers is not in the document]
Comment (yangw-dev) [reply torchci/lib/bot/autoLabelBot.ts:24]: I can clean the list here, just want to confirm:
remove ciflow/periodic  from  IssueTitleRegexToLabel
remove DISABLED, UNSTABLE from PrTitleRegexToLabel

File: torchci/lib/bot/autoLabelBot.ts [hunk 3]
```diff
@@ -12,14 +12,25 @@ import {
...
 } from "./utils";
 
-const titleRegexToLabel: [RegExp, string][] = [
+// List of regex patterns for assigning labels to both Pull Requests and Issues
+const IssueAndPRRegexToLabel: [RegExp, string][] = [
   [/rocm/gi, "module: rocm"],
   [/rocm/gi, "ciflow/rocm"],
   [/vulkan/gi, "module: vulkan"],
-  [/vulkan/gi, "ciflow/periodic"], // Vulkan tests are run periodically
   [/DISABLED\s+test.*\(.*\)/g, "skipped"],
```
Comment (clee2000) [inline torchci/lib/bot/autoLabelBot.ts:20]: disabled should be in the Issue title list, ciflow/rocm should be in the pr title list, and I think Huy mentioned that he got rid of vulkan

Comment (yangw-dev) [reply torchci/lib/bot/autoLabelBot.ts:20]: there are two DISABLEDs, assume you say that all of them should move to Issue title list?

**Review by clee2000 — APPROVED** [review]

**Review by huydhn — APPROVED** [review]

**Pull request merged**